pkgs/development/compilers/lunarml/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, fetchFromGitHub
+, fetchzip
+, stdenvNoCC
+, mlton
+, lua5_3
+, luajit
+, nodejs
+, withExamples ? false
+}:
+
+let
+  pname = "lunarml";
+in
+stdenvNoCC.mkDerivation {
+  inherit pname;
+
+  version = "unstable-2022-01-24";
+
+  src = fetchFromGitHub {
+    owner = "minoki";
+    repo = "LunarML";
+    rev = "60b646ae7d55550626642a1b3f9a0af3944cc2a6";
+    sha256 = "sha256-lI947OQlb5s9tkSns9bP4uNyjsa1fsPuPiczgfVRe1I=";
+  };
+
+  nativeBuildInputs = [
+    mlton
+  ];
+
+  nativeCheckInputs = [
+    lua5_3
+    luajit
+    nodejs
+  ];
+
+  postPatch = ''
+    cd "thirdparty"
+    make smlnj-lib
+    make ml-yacc
+    cd ".."
+  '';
+
+  postBuild = ''
+    cd "thirdparty"
+    make install-smlnj-lib
+    make install-mlyacc-lib
+    cd ".."
+  '';
+
+  doCheck = true;
+
+  installPhase = ''
+    mkdir -p $out/{bin,lib,share/doc/${pname}}
+    cp -r bin lib $out
+    cp -r doc/* README.* LICENSE* $out/share/doc/${pname}
+  '' + lib.optionalString withExamples ''
+    cp -Tr example $out/share/doc/${pname}/example
+  '';
```

Comment:
Since `withExamples` does not affect the actual build, I think this would be better suited for a [split-output derivation](https://nixos.org/manual/nixpkgs/unstable/#sec-multiple-outputs-). Unless there's a good reason to have the examples separate from the rest of the doc, I'd just make a `doc` output with all of `/share/doc`.

Reply:
Add `"doc"` to `outputs`, remove `withExamples`, use `$doc/share/doc` instead of `$out/share/doc`